packages/hyperion-dom/src/EventHandlerAttributeInterceptor.ts

```diff
@@ -2,23 +2,54 @@
  * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
  */
 
-import { AttributeInterceptor, getAttributeInterceptor } from "@hyperion/hyperion-core/src/AttributeInterceptor";
+import { AttributeInterceptor, interceptAttributeBase } from "@hyperion/hyperion-core/src/AttributeInterceptor";
 import { ShadowPrototype } from "@hyperion/hyperion-core/src/ShadowPrototype";
+import { ElementAttributeInterceptor } from "./ElementAttributeInterceptor";
+
 
 class EventHandlerAttributeInterceptor<
   BaseType extends EventTarget & { [key: string]: any },
   Name extends string,
-  > extends AttributeInterceptor<BaseType, Name> {
+> extends AttributeInterceptor<BaseType, Name> {
 
 }
 
+/**
+ * Use this function when even even handler cannot be also added via
+ * a setAttribute or Attr.value on the element.
+ * This usually applies to non-element nodes.
+ */
 export function interceptEventHandlerAttribute<
   BaseType extends EventTarget & { [key: string]: any },
   Name extends string,
-  >(
-    name: Name,
-    shadowPrototype: ShadowPrototype<BaseType>
-  ): EventHandlerAttributeInterceptor<BaseType, Name> {
-  const desc = getAttributeInterceptor<BaseType, Name, BaseType[Name], BaseType[Name], EventHandlerAttributeInterceptor<BaseType, Name>>(name, shadowPrototype);
-  return desc?.interceptor ?? new EventHandlerAttributeInterceptor(name, shadowPrototype, desc);
-}
\ No newline at end of file
+>(
+  name: Name,
+  shadowPrototype: ShadowPrototype<BaseType>
+): EventHandlerAttributeInterceptor<BaseType, Name> {
+  return interceptAttributeBase<BaseType, Name, BaseType[Name], BaseType[Name], EventHandlerAttributeInterceptor<BaseType, Name>>(name, shadowPrototype, EventHandlerAttributeInterceptor);
+}
+
+class ElementEventHandlerAttributeInterceptor<
+  BaseType extends Element & { [key: string]: any },
+  Name extends string,
+> extends ElementAttributeInterceptor<BaseType, Name, BaseType[Name], BaseType[Name]> {
+
+}
+
+/**
+ * Use this function when the event handler can be set via node.onX
+ * as well as node.setAttribute("X", ....) or node.getAttributeNode("X").value = ...
+ * This mostly applies to Elements. 
+ * In these cases, one need to cover both raw (text) and processed (function) values
+ * for the handler. 
+ */
+export function interceptElementEventHandlerAttribute<
```

Comment:
This is the more advanced case where someone could add `setAttribute("onclick", ....) to an element, but it very rare and would need to further test the implementation. So for now, keep the old simple one.